source/Background/utils.js

```diff
@@ -106,9 +107,108 @@ export const fetchCanistersInfo = async (whitelist) => {
 // TokenIdentifier is SYMBOL or  CanisterID
 // Return ICP by default
 export const getToken = (tokenIdentifier, assets) => {
-  if (!tokenIdentifier) return assets.filter(asset => asset.canisterId === ICP_CANISTER_ID)[0]
+  if (!tokenIdentifier) return assets.filter((asset) => asset.canisterId === ICP_CANISTER_ID)[0];
+
+  if (validateCanisterId(tokenIdentifier)) {
+    return assets.filter((asset) => asset.canisterId === tokenIdentifier)[0];
+  }
+
+  return assets.filter((asset) => asset.symbol === tokenIdentifier)[0];
+};
+export const bufferToBase64 = (buf) => Buffer.from(buf).toString('base64');
+
+export const base64ToBuffer = (base64) => Buffer.from(base64, 'base64');
 
-  if (validateCanisterId(tokenIdentifier)) return assets.filter(asset => asset.canisterId === tokenIdentifier)[0]
+export const isDeepEqualObject = (obj1, obj2) => JSON.stringify(obj1) === JSON.stringify(obj2);
+
+export const validateBatchTx = (savedTxInfo, canisterId, methodName, arg) => {
+  if (!savedTxInfo
+    || savedTxInfo.canisterId !== canisterId
+    || savedTxInfo.methodName !== methodName) {
+    return false;
+  }
+
+  if (savedTxInfo.args) {
+    const decodedArguments = recursiveParseBigint(
+      PlugController.IDLDecode(arg),
+    );
+
+    const allArgsEqual = savedTxInfo.args.every(
+      (savedTxArgs, index) => isDeepEqualObject(savedTxArgs, decodedArguments[index]),
+    );
+
+    if (!allArgsEqual) {
+      return false;
+    }
+  }
 
-  return assets.filter(asset => asset.symbol === tokenIdentifier)[0]
-}
+  return true;
+};
+
+export const handleCallRequest = async ({
```

Comment:
This doesnt seem to belong here, I'd put it in the transactions module right?

Reply:
Not sure, I would put only RPC handlers into the modules